### DISCUSSION TITLE
[BUG] Impossible to enter WLAN credentials without WebUI ?

### DISCUSSION THREAD

**Discussion by probonopd** [luc-github/ESP3D#920] [F.A.Q] [answered]
**Describe the bug**

How can I enter my WLAN credentials?

Only after the correct index.html.gz has been uploaded.
This is cumbersome. Chicken and egg problem.
I am using my smartphone to configure ESP devices to join my WLAN network.
Once the devices are on the WLAN network, I can then use the computer to access them.

**To Reproduce**

1. Flash the firmware on a new ESP32-CAM
2. Power on the device
3. Using your smartphone, connect to the AP (password: 12345678 - why a password at all if everyone knows it? Unnecessary extra hassle)
4. Try to configure the device to use your real WLAN in station mode - it is impossible to do! (You would have to download `index.html.gz` to your smartphone and upload it from there... too cumbersome. I want to do this from my desktop PC which has no WLAN. Only Ethernet. So I need to bring the ESP device into station mode first, before I can access it from the PC.
 
**Expected behavior**

Can set WLAN SSID and password without having to upload `index.html.gz` first.

**ESP3D Firmware:**

3.0.0

**Comment by luc-github** [comment, marked as answer]
To enter WLAN credentials without WebUI there are at least 2 methods:

1 - You can preset the information in configuration.h so you will directly connect to you access point at boot.
https://github.com/luc-github/ESP3D/blob/3.0/esp3d/configuration.h#L26-L37

2 - You can use the [direct commands](http://esp3d.io/esp3d/v3.x/documentation/commands/index.html) using terminal tab in embedded / maintenance page
![image](https://user-images.githubusercontent.com/8822552/232637735-35472b59-26da-41a7-8870-69ecc3a23d71.png)


>, connect to the AP (password: 12345678 - why a password at all if everyone knows it? Unnecessary extra hassle)

Because even with simple password, that I am sure not anyone knows because even on doc people ask for it ..., is still better than none and ultimatly because I decided it will be like this ^_^
